src/pages/scripting/how-it-works/index.md

```diff
@@ -0,0 +1,100 @@
+
+
+### Global Await
+`await` expressions cause `async` function execution to pause until a Promise is either fulfilled or rejected, and to resume execution
+of the `async` function after fulfillment. When resumed, the value of the `await` expression is that of the fulfilled Promise.
+
+Global await means awaiting for a global scope async function to finish. 
+
+```js
+function anyAsyncFunction() {
+    return new Promise((resolve, reject) => {
+        const val = Math.random();
+        if (val > 0.5) {
+            console.log("Resolve promise");
+            resolve(val);
+        } else {
+            console.error("Rejecting promise");
+            reject(val);
+        }
+    })
+}
+
+// Global await
+try {
+    const val = await anyAsyncFunction();
+    console.log("Value from function", val);
+} catch {
+    console.log("Rejected promise!;")
+}
+console.log("Script completed");
+```
+
+## Permitted UXP Modules
+With plugin development, developers define which UXP modules they want to access in the `manifest.json` file. Script permissions are managed by Photoshop internally and no manifest configuration is required. 
+
+**At this time, not all UXP modules are accessible by scripts but we plan on enabling more modules in future UXP versions.**  All modules that are currently supported are enabled by default:
+
+
+| UXP Module | Supported | Current Access Level | Sample
+| --- | --- | --- | --- | 
+| Fonts | Yes | Installed fonts can be read | [View sample.](../samples/index.md#access-installed-fonts) |
+| Clipboard | Yes | Read/write access | [View sample.](../samples/index.md#readwrite-to-clipboard)| 
+| LocalFileSystem | Yes | File pickers can be used to open/save files. You can also call the ```getTemporaryFolder API``` to access a temporary data folder. | [View sample.](../samples/index.md#access-the-local-filesystem) |
+| Network | No |  |
+| LaunchProcess | No |  |
+| WebView | No |  |
+| IPC  | No |  |
+
+If you need to use the UXP modules that are not yet enabled for scripts, you should opt to create a UXP plugin instead. 
+
+## FAQ
+* What are the required minimum versions for UDT (for debugging) and Photoshop?
+  * UXP Scripting is available in Photoshop v23.5 and UDT v1.6.
+* Can you invoke a script from within another script? 
+  * No, inter-script communication is not possible. 
+* Can you pass arguments to scripts? 
+  * For this first release, passing arguments is not supported but **will be possible in a future release.**
+* Can scripts be executed from plugins? 
+  * No; however, any UXP script code should be able to run from within a UXP plugin.
+* Can I enable permissions for a module?
+  * Developers cannot enable or seek the permission for a module. All permitted modules are enabled by default. 
```

Comment:
Can we change _All permitted modules are enabled by default_ to 'Modules are enabled by the host application' ?

Reply:
While I agree with all your other comments, I respectfully disagree on this change request. "Host apps" is, to me, very much "Adobe-internal talk" in the sense that as a plugin developer, I don't care about "host apps", I just want to "write a Photoshop script". The burden of understanding how UXP is integrated into various Creative Cloud apps should be on us, not the plugin- and even less on script developers / writers.

If we really have to change it, let's write "Photoshop" instead of "host application". But in my opinion, this info might be relevant for us, internally (i.e., which team is responsible for which modules are enabled), but this shouldn't be the concern of third-party developers. In other words: As a developer, I care about what's enabled and what is not (and maybe, if it affects me, technical reasons), but not the organizational cause of it.

So in my opinion, we shouldn't really change this, but if we have to, let's at least avoid the "host application" wording 😉.

Reply:
Sure. Lets avoid  the term 'host application' all together. I was just trying to point out that 'All permitted modules are enabled by default' could be misunderstood as 'All modules are permitted/enabled by default'.

Reply:
I agree with Pablo's comment here but think we can elaborate on Padma's change as well- what I'll do is change it to "By default, all permitted modules are enabled by Photoshop."